media-video/pipewire/pipewire-0.3.25.ebuild

```diff
@@ -0,0 +1,234 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit meson optfeature
+
+if [[ ${PV} == 9999 ]]; then
+	EGIT_REPO_URI="https://gitlab.freedesktop.org/${PN}/${PN}.git"
+#	EGIT_BRANCH="master"
+	inherit git-r3
+else
+	SRC_URI="https://gitlab.freedesktop.org/${PN}/${PN}/-/archive/${PV}/${P}.tar.gz"
+	KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~x86"
+fi
+
+DESCRIPTION="Multimedia processing graphs"
+HOMEPAGE="https://pipewire.org/"
+
+LICENSE="LGPL-2.1+"
+SLOT="0/0.3"
+IUSE="aac aptx bluetooth doc extra gstreamer jack-client ldac pipewire-alsa systemd test v4l"
+
+# Once replacing system JACK libraries is possible, it's likely that
+# jack-client IUSE will need blocking to avoid users accidentally
+# configuring their systems to send PW sink output to the emulated
+# JACK's sink - doing so is likely to yield no audio, cause a CPU
+# cycles consuming loop (and may even cause GUI crashes)!
+
+REQUIRED_USE="
+	aac? ( bluetooth )
+	aptx? ( bluetooth )
+	ldac? ( bluetooth )
+"
+
+BDEPEND="
+	app-doc/xmltoman
+	doc? (
+		app-doc/doxygen
+		media-gfx/graphviz
+	)
+"
+RDEPEND="
+	media-libs/alsa-lib
+	sys-apps/dbus
+	sys-libs/ncurses[unicode]
+	virtual/libudev
+	bluetooth? (
+		aac? ( media-libs/fdk-aac )
+		aptx? ( media-libs/libopenaptx )
+		ldac? ( media-libs/libldac )
+		media-libs/sbc
+		>=net-wireless/bluez-4.101:=
+	)
+	extra? (
+		>=media-libs/libsndfile-1.0.20
+	)
+	gstreamer? (
+		>=dev-libs/glib-2.32.0:2
+		>=media-libs/gstreamer-1.10.0:1.0
+		media-libs/gst-plugins-base:1.0
+	)
+	jack-client? ( >=media-sound/jack2-1.9.10:2[dbus] )
+	pipewire-alsa? (
+		>=media-libs/alsa-lib-1.1.7
+		|| (
+			media-plugins/alsa-plugins[-pulseaudio]
+			!media-plugins/alsa-plugins
+		)
+	)
+	!pipewire-alsa? ( media-plugins/alsa-plugins[pulseaudio] )
```

Comment:
Hello,

I'm wondering why this is needed. I currently have a setup where all application use exclusively pulseaudio, so I don't need or have pulseaudio plugin for ALSA installed. It works perfectly for many years. 

Can this be removed? What if user do not need any plugin for ALSA, neither pusle nor pipewire?

Reply:
It would break ALSA applications for great many people - if you don't want the pulseaudio's plugin (which does not pull in pulseaudio itself), then you can set the pipewire-alsa plugin and the native one will be installed. ;)

Reply:
Hm, media-sound/pulseaudio has recommended 'alsa-plugin' USE flag for this. It should up to the user whether to enable it or no.

I have it disabled and everything works perfectly. I think media-video/pipewire can do the same and install ALSA plugin only if requested by USE flag (be it plugin from PulseAudio or from Pipewire).

Reply:
But if you try to use an ALSA application, it will not work. Gentoo generally is not in the business of making dysfunctional or known broken packages for the sake of it - there has to be a technical reason for this. The ALSA plugin is generally a good fallback to have for any program that either lacks a newer interface or has been configured to use the ALSA API, which is a perfectly valid choice. If you still disagree, open a formal [Gentoo bug](https://bugs.gentoo.org).

Reply:
Done https://bugs.gentoo.org/785076 :)